containers/src/Data/IntMap/Internal.hs

```diff
@@ -353,26 +356,22 @@ intFromNat = fromIntegral
 
 -- See Note: Order of constructors
 data IntMap a = Bin {-# UNPACK #-} !Prefix
-                    {-# UNPACK #-} !Mask
```

Comment:
Somewhere (probably around here, but it could be in a separate file if you'd prefer) we need an explicit discussion of the relationship between the new representation and the way it's done in the paper.

Reply:
Done.